core-bundle/src/EventListener/DataContainer/PageUrlListener.php

```diff
@@ -223,7 +215,9 @@ public function purgeSearchIndex(int $pageId): void
         ;
 
         foreach ($urls as $url) {
-            $this->searchIndexer->delete(new Document(new Uri($url), 200));
+            /** @var Search $search */
+            $search = $this->framework->getAdapter(Search::class);
```

Comment:
These two lines should be outside the `foreach` loop.

Reply:
not really related to this PR, but changed in 97e747ce07316253ed444fcbec25ca12884753da 😆